specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/stable/2020-03-01/functions.json

```diff
@@ -0,0 +1,811 @@
+        "description": "Retrieves the default definition of a function based on the parameters specified.",
+        "x-ms-examples": {
+          "Retrieve the default definition for an Azure ML function": {
+            "$ref": "./examples/Function_RetrieveDefaultDefinition_AzureML.json"
+          }
+        },
+        "parameters": [
+          {
+            "name": "functionRetrieveDefaultDefinitionParameters",
+            "in": "body",
+            "required": false,
+            "schema": {
+              "$ref": "#/definitions/FunctionRetrieveDefaultDefinitionParameters"
+            },
+            "description": "Parameters used to specify the type of function to retrieve the default definition for."
+          },
+          {
+            "$ref": "../../common/v1/definitions.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../common/v1/definitions.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../common/v1/definitions.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../common/v1/definitions.json#/parameters/StreamingJobNameParameter"
+          },
+          {
+            "$ref": "#/parameters/FunctionNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully retrieved the function's default definition.",
+            "schema": {
+              "$ref": "#/definitions/Function"
+            }
+          },
+          "default": {
+            "description": "Error.",
+            "schema": {
+              "$ref": "../../common/v1/definitions.json#/definitions/Error"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Function": {
+      "description": "A function object, containing all information associated with the named function. All functions are contained under a streaming job.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/SubResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/FunctionProperties",
+          "description": "The properties that are associated with a function."
+        }
+      }
+    },
+    "FunctionProperties": {
+      "description": "The properties that are associated with a function.",
+      "discriminator": "type",
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "Indicates the type of function."
+        },
+        "etag": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The current entity tag for the function. This is an opaque string. You can use it to detect whether the resource has changed between requests. You can also use it in the If-Match or If-None-Match headers for write operations for optimistic concurrency."
+        }
+      },
+      "required": [
```

Comment:
>required [](http://example.com/codeflow?start=7&length=8)

Changing a property to required is a new breaking change. Was it covered by your breaking change review/approval?

Reply:
I changed these to required because I was getting a validation error that said all discriminator values must be required.

Reply:
@mentat9 I changed these properties to required because the validation tool was throwing an error saying that discriminator values must be required -- is this not the case? If not, I can change it back